streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java

```diff
@@ -284,10 +278,21 @@ public void run() {
         }
         setState(State.RUNNING);
 
+        boolean wipeStateStore = false;
         try {
             while (stillRunning()) {
                 stateConsumer.pollAndUpdate();
             }
+        } catch (final InvalidOffsetException recoverableException) {
+            wipeStateStore = true;
+            log.error(
+                "Updating global state failed. You can restart KafkaStreams to recover from this error.",
```

Comment:
```suggestion
                "Updating global state failed due to inconsistent local state. Will attempt to clean up the local state. You can restart KafkaStreams to recover from this error.",
```

Just a thought to indicate why just restarting would recover anything.